utils/plot.hpp

```diff
@@ -0,0 +1,184 @@
+#ifndef PLOT_HPP
+#define PLOT_HPP
+
+#define PY_SSIZE_T_CLEAN
+#include <Python.h>
+#include <string>
+
+int scatter(const std::string fname, const std::string type,const std::string figTitle, const int figWidth = 26, const int figHeight = 7) {
```

Comment:
```suggestion
int scatter(const std::string& fname, const std::string& type, const std::string& figTitle, const int figWidth = 26, const int figHeight = 7)
{
```

Just to match the mlpack style and to avoid an extra copy.